hashtable/hashtable.py

```diff
@@ -40,22 +59,38 @@ def hash_index(self, key):
         return self.djb2(key) % self.capacity
 
     def put(self, key, value):
-        """
-        Store the value with the given key.
 
-        Hash collisions should be handled with Linked List Chaining.
+        index = self.hash_index(key)
+        if self.storage[index] is None:
+            self.storage[index] = HashTableEntry(key, value)
+            self.entries += 1
+        else:
+            new_node = HashTableEntry(key, value)
+            new_node.next = self.storage[index]
+            self.storage[index] = new_node
+            self.entries += 1
 
-        Implement this.
-        """
 
     def delete(self, key):
-        """
-        Remove the value stored with the given key.
 
-        Print a warning if the key is not found.
+        index = self.hash_index(key)
+        prev = None
+        cur = self.storage[index]
 
-        Implement this.
-        """
+        #if there's nothing in that storage
+
+        if cur.key == key:
+            self.storage[index] = cur.next
+            return self.get(key)
```

Comment:
I would check the order here. Are you able to get the key after you set self.storage[key] to cur.next?

Reply:
Same thought for lines 90 and 93